packages/medusa/src/services/discount-condition.ts

```diff
@@ -0,0 +1,233 @@
+import { MedusaError } from "medusa-core-utils"
+import { BaseService } from "medusa-interfaces"
+import { EntityManager } from "typeorm"
+import { EventBusService } from "."
+import { Discount, DiscountCondition, DiscountConditionType } from "../models"
+import { DiscountRepository } from "../repositories/discount"
+import { DiscountConditionRepository } from "../repositories/discount-condition"
+import { DiscountRuleRepository } from "../repositories/discount-rule"
+import { FindConfig } from "../types/common"
+import { UpsertDiscountConditionInput } from "../types/discount"
+import { PostgresError } from "../utils/exception-formatter"
+
+/**
+ * Provides layer to manipulate discount conditions.
+ * @implements {BaseService}
+ */
+class DiscountConditionService extends BaseService {
+  protected readonly manager_: EntityManager
+  protected readonly discountRuleRepository_: typeof DiscountRuleRepository
+  protected readonly discountRepository_: typeof DiscountRepository
+  protected readonly discountConditionRepository_: typeof DiscountConditionRepository
+  protected readonly eventBus_: EventBusService
+
+  constructor({
+    manager,
+    discountRuleRepository,
+    discountRepository,
+    discountConditionRepository,
+    eventBusService,
+  }) {
+    super()
+
+    /** @private @const {EntityManager} */
+    this.manager_ = manager
+
+    /** @private @const {DiscountRuleRepository} */
+    this.discountRuleRepository_ = discountRuleRepository
+
+    /** @private @const {DiscountRepository} */
+    this.discountRepository_ = discountRepository
+
+    /** @private @const {DiscountConditionRepository} */
+    this.discountConditionRepository_ = discountConditionRepository
+
+    /** @private @const {EventBus} */
+    this.eventBus_ = eventBusService
+  }
+
+  withTransaction(transactionManager: EntityManager): DiscountConditionService {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new DiscountConditionService({
+      manager: transactionManager,
+      discountRuleRepository: this.discountRuleRepository_,
+      discountRepository: this.discountRepository_,
+      discountConditionRepository: this.discountConditionRepository_,
+      eventBusService: this.eventBus_,
+    })
+
+    cloned.transactionManager_ = transactionManager
+
+    return cloned
+  }
+
+  async retrieve(
+    conditionId: string,
+    config: FindConfig<DiscountCondition> = {}
+  ): Promise<DiscountCondition> {
+    return await this.atomicPhase_(async (manager: EntityManager) => {
+      const conditionRepo = manager.getCustomRepository(
+        this.discountConditionRepository_
+      )
+
+      const query = this.buildQuery_({ id: conditionId }, config)
+      const condition = await conditionRepo.findOne(query)
```

Comment:
since a discount condition cannot exists without a discount, shouldn't we add an inner join on the discount to ensure to look for the condition on the discount we are currently working on?